### PR TITLE
Turn down graphic settings when unfocused & improve Turkish language translation

### LINKED ISSUE #54: suggestion: turn down graphical settings in background
would it be viable to make minecraft render at lower settings in the background to lower use even more? (for example: turn graphics down from fabulous down to fast) I think if there is not much difference, maybe it would allow to set the FPS limit to something higher like 15, so that you can still see what is happening, just with reduced quality instead of reduced frame rates.

### PR DESCRIPTION
**Pull request by EuropaYou**
Will resolve #54 

For now; turns off Entity Shadows, Smooth Lighting and clouds; Minimizes Particles; Sets Graphics to Fast.

Let me know if you want to add/remove features.

### CODE REVIEW

**Review by LostLuma — CHANGES_REQUESTED** [review]
Hello, thanks for the PR! Since this causes the entire world to reload every time the window is (un)focused I don't think this should be enabled by default, since the flashing from reloading and the resource usage jump on re-focusing is not expected of a resource saving mod IMO. Otherwise this looks good as-is I think :)

File: src/main/java/dynamicfps/DynamicFPSConfig.java [hunk 1]
```diff
@@ -19,17 +19,18 @@ public final class DynamicFPSConfig {
 	public float unfocusedVolumeMultiplier = 0.25f;
 	/// Volume multiplier when not visible.
 	public float hiddenVolumeMultiplier = 0f;
+	public boolean reduceGraphicsWhenUnfocused = true;
```
Comment (LostLuma) [inline src/main/java/dynamicfps/DynamicFPSConfig.java:22]:
```suggestion
	public boolean reduceGraphicsWhenUnfocused = false;
```

**Comment by juliand665** [comment]
I wonder if it would make sense to separate out the settings changes that require a world reload into a separate setting? I assume fast graphics is the big one here.

**Comment by LostLuma** [comment]
> I assume fast graphics is the big one here.

Both graphics and lighting quality changes cause the world to reload sadly.

I also got around to testing this now and looking at other options that could be useful, there's also the option to decrease entity distance when unfocused, which might be good to add to this PR either way but also to the non-reloading "safe" options to change list.

File: src/main/java/dynamicfps/DynamicFPSMod.java [hunk 2]
```diff
@@ -104,44 +144,56 @@ private static void checkForStateChanges() {
...
+
 	private static void onUnfocus() {
+		GameOptions gameOptions = MinecraftClient.getInstance().options;
+		if(config.fullyReduceGraphicsWhenUnfocused){
+			reduceGraphicsFully(gameOptions);
+		}
+		if(config.reduceGraphicsWhenUnfocused) {
+			reduceGraphics(gameOptions);
+		}
+
```
Comment (LostLuma) [inline src/main/java/dynamicfps/DynamicFPSMod.java:163]:
```suggestion
		if (config.fullyReduceGraphicsWhenUnfocused) {
			reduceGraphicsFully(gameOptions);
		}
		if (config.reduceGraphicsWhenUnfocused) {
			reduceGraphics(gameOptions);
		}
```

File: src/main/java/dynamicfps/DynamicFPSMod.java [hunk 3]
```diff
@@ -104,44 +144,56 @@ private static void checkForStateChanges() {
 			}
 		}
 	}
-	
+
 	private static void onFocus() {
 		setVolumeMultiplier(1);
+		GameOptions gameOptions = MinecraftClient.getInstance().options;
+		increaseGraphicsFully(gameOptions);
+		increaseGraphics(gameOptions);
```
Comment (LostLuma) [inline src/main/java/dynamicfps/DynamicFPSMod.java:152]: These methods can not be called unconditionally, as all of the `previousX` fields have not necessarily been initialized (or rather, are the default, so eg. boolean values are `false`) when the options are disabled, causing these options to be set to erroneous values.
```suggestion
		if (config.fullyReduceGraphicsWhenUnfocused) {
			increaseGraphicsFully(gameOptions);
		}
		if (config.reduceGraphicsWhenUnfocused) {
			increaseGraphics(gameOptions);
		}
```

**Review by LostLuma — CHANGES_REQUESTED** [review]

**Comment by LostLuma** [comment]
I'm not sure which bug you're referencing (unless it's the one I commented on in the last review, but that is solved now?), from my side this looks fine now. Please let me know if you'd still like to change something @EuropaYou.

**Review by LostLuma — APPROVED** [review]

**Comment by LostLuma** [comment]
> Oh, I'm so sorry. I've been very busy the last two days.

No worries as I said!

> I also get these errors when the window regains focus:

Those are also caused by what I mentioned [in my last review](https://github.com/juliand665/Dynamic-FPS/pull/96#discussion_r1285295894). I assume you did not build the jar again with the two suggested commits included, meaning you don't have the fix. You can grab a [build from CI](https://github.com/juliand665/Dynamic-FPS/actions) instead if you'd like (a new proper release will still take a bit, have more PRs to write 🙂).

**Pull request merged**